src/unused_finder/node_visitor.rs

```diff
@@ -169,26 +177,42 @@ impl UnusedFinderVisitor {
             _ => {}
         });
     }
+
+    pub fn has_disable_export_comment(&self, lo: BytePos) -> bool {
+        if let Some(comments) = self.comments.get_leading(lo) {
+            return comments.iter().any(|c| {
+                c.kind == CommentKind::Line && c.text.trim().starts_with("@ALLOW-UNUSED-EXPORT")
+            });
+        }
+        false
+    }
 }
 
-impl Visit for UnusedFinderVisitor {
+impl Visit for ExportsCollector {
     // Handles `export default foo`
     fn visit_export_default_expr(&mut self, expr: &ExportDefaultExpr) {
         expr.visit_children_with(self);
-        self.exported_ids.insert(ExportedItem::Default);
+        if !self.has_disable_export_comment(expr.span_lo()) {
+            self.exported_ids.insert(ExportedItem::Default);
+        }
     }
 
     /**
      * Handles scenarios where `export default` has an inline declaration, e.g. `export default class Foo {}` or `export default function foo() {}`
      */
     fn visit_export_default_decl(&mut self, decl: &ExportDefaultDecl) {
         decl.visit_children_with(self);
-        self.exported_ids.insert(ExportedItem::Default);
+        if !self.has_disable_export_comment(decl.span_lo()) {
+            self.exported_ids.insert(ExportedItem::Default);
```

Comment:
to confirm: in a follow-up change you're going to keep adding the exported_ids, but tracking if they were ignored

Reply:
Yes, will add struct for the metadata on exported items and validate if they're allowed to be unused during the import/export validation process.